pkg/commands/kustomizationfile.go

```diff
@@ -129,20 +144,20 @@ func stringInSlice(str string, list []string) bool {
 func (mf *kustomizationFile) parseCommentedFields(content []byte) error {
 	buffer := bytes.NewBuffer(content)
 	var comments [][]byte
-	var currentfield string
 
 	line, err := buffer.ReadBytes('\n')
 	for err == nil {
 		if isCommentOrBlankLine(line) {
 			comments = append(comments, line)
-		} else if recognizedFields.Match(line) {
-			fields := recognizedFields.FindSubmatch(line)
-			currentfield = string(fields[1])
-			mf.originalFields = append(mf.originalFields, &commentedField{field: currentfield, comment: bytes.Join(comments, []byte(``))})
-			comments = [][]byte{}
-		} else if len(comments) > 0 {
-			mf.originalFields[len(mf.originalFields)-1].appendComment(bytes.Join(comments, []byte(``)))
-			comments = [][]byte{}
+		} else {
+			matched, field := findMatchedField(line)
+			if matched {
+				mf.originalFields = append(mf.originalFields, &commentedField{field: field, comment: bytes.Join(comments, []byte(``))})
```

Comment:
Please add some line feeds and use a helper to increase readbility.

eg 
```
func squash(x [][]byte) []byte {
  return bytes.Join(x, []byte(``))
}
```

Reply:
sounds good. Will add it.

Reply:
done